### PR TITLE
Custom error handling sample

### PR DESCRIPTION
**Pull request by HEskandari**

### CODE REVIEW

File: samples/message-error-handling/Core_4/CustomErrorHandling/CustomFaultManager.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+using NServiceBus;
+using NServiceBus.Config;
+using NServiceBus.Config.ConfigurationSource;
+using NServiceBus.Faults;
+using NServiceBus.Logging;
+using NServiceBus.Transports;
+
+public class CustomFaultManager : IManageMessageFailures
+{
+    readonly ISendMessages sender;
```
Comment (SimonCropp) [inline samples/message-error-handling/Core_4/CustomErrorHandling/CustomFaultManager.cs]: remove the readonly

File: samples/message-error-handling/Core_4/CustomErrorHandling/MyMessageHandler.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    public MyMessageHandler(IBus bus)
+    {
+        this.bus = bus;
+    }
+
+    public void Handle(MyMessage message)
+    {
+        if(message.ThrowCustomException)
+            throw new MyCustomException();
```
Comment (SimonCropp) [inline samples/message-error-handling/Core_4/CustomErrorHandling/MyMessageHandler.cs]: add missing braces

File: samples/message-error-handling/Core_6/CustomErrorHandling/Program.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        AsyncMain().GetAwaiter().GetResult();
+    }
+
+    static async Task AsyncMain()
+    {
+        Console.Title = "Samples.CustomErrorHandling";
+        LogManager.Use<DefaultFactory>()
+            .Level(LogLevel.Warn);
+
+        var configure = new EndpointConfiguration("Samples.CustomErrorHandling");
```
Comment (SimonCropp) [inline samples/message-error-handling/Core_6/CustomErrorHandling/Program.cs]: name the variable `endpointConfiguration`

File: samples/message-error-handling/Core_6/CustomErrorHandling/Program.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        configure.UseSerialization<JsonSerializer>();
+        configure.UsePersistence<InMemoryPersistence>();
+        configure.EnableInstallers();
+        configure.SendFailedMessagesTo("error");
+
+        var r = configure.Recoverability();
```
Comment (SimonCropp) [inline samples/message-error-handling/Core_6/CustomErrorHandling/Program.cs]: name the variable `recoverability`

**Comment by HEskandari** [comment]
This is still a WIP, I need to add the page to describe the sample.

**Comment by HEskandari** [comment]
@Particular/docs-maintainers Please review. First time using partials, please double check to make sure there's no errors.

**Comment by SimonCropp** [comment]
@HEskandari thanks

**Comment by HEskandari** [comment]
Thank you @SimonCropp :)

**Pull request merged**